app/src/main/java/com/bignerdranch/android/rickandmorty/navigation/ListFragment.kt

```diff
@@ -0,0 +1,141 @@
+import kotlin.properties.Delegates
+
+class ListFragment : Fragment() {
+
+    private var _binding: FragmentListBinding? = null
+    private val binding get() = requireNotNull(_binding)
+    private var pageCount by Delegates.notNull<Int>()
+    private val adapter by lazy {
+        PersonAdapter {
+            findNavController().navigate(
+                R.id.action_listFragment_to_infoFragment,
+                bundleOf(
+                    InfoFragment.KEY_PERSON to it.id,
+                    InfoFragment.KEY_PERSON_NAME to it.name
+                )
+            )
+        }
+    }
+
+    private var isLoading = false
+    private var count = 1
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        return FragmentListBinding.inflate(inflater, container, false)
+            .also { _binding = it }
+            .root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        // Loading the first 20 items
+        if (count == 1) {
+            loadRickAndMortyCharacters(count)
+            count++
+        }
+
+        val linearLayout = LinearLayoutManager(requireContext())
+
+        with(binding) {
+            recyclerView.layoutManager = linearLayout
+            recyclerView.adapter = adapter
+            recyclerView.addSpaceDecoration(SPACE_DECORATION)
+            recyclerView.addPaginationScrollListener(linearLayout, ITEM_TO_LOAD_QUANTITY) {
+                if (!isLoading && count <= pageCount) {
+                    loadRickAndMortyCharacters(count)
+                }
+            }
+
+            swipeRefresh.setOnRefreshListener {
+                count = 1
+                val listPersonEmpty = listOf<PersonItem.Person>()
+                adapter.submitList(listPersonEmpty)
+                loadRickAndMortyCharacters(count)
+                count++
+                showToast("Data updated")
+                swipeRefresh.isRefreshing = false
+            }
+        }
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
+
+    private fun loadRickAndMortyCharacters(page: Int) {
+        val load = RickAndMortyService.getInstance().getAllCharacters(page)
+        isLoading = true
+        load.enqueue(object : Callback<ListPerson> {
+            override fun onResponse(call: Call<ListPerson>, response: Response<ListPerson>) {
+                if (response.isSuccessful) {
+                    val listPerson = response.body()
+
+                    if (listPerson != null) {
+                        pageCount = listPerson.info.pages
+                    }
+
+                    val currentList = adapter.currentList.toList().dropLast(1)
+                    val listResult = currentList
+                        .plus(listPerson?.results as List<PersonItem.Person>)
```

Comment:
кастовать nullable объект к листу опасно, лучше так
```suggestion
                        .plus(listPerson?.results ?: emptyList())
```